### ISSUE TITLE
Dynamic STRING variable's address of is not working

### ISSUE THREAD

**Issue by orlof** [neilsf/xc-basic3#187]
The problem is visible in `print @s, @a`, which prints "0 0".

(It was also a bit surprising that I didn't find a's value anywhere in frame stack when I tried to search for it with the FOR-loop, but that may well be my misunderstanding on how strings work. It just made me wonder if problem is not the address-of operator, but reservation of string space from frame stack?)

```
INCLUDE "hex.bas"

SUB test()
    DIM s AS STRING * 4: s = "abcd"
    DIM a AS WORD: a = $beef
    print @s, @a

    FOR t AS WORD = 256*PEEK(16)+PEEK(15) TO $cf00
        print hex(t), hex(peek(t))
    NEXT t
END SUB

CALL test()
```

<img width="652" alt="image" src="https://user-images.githubusercontent.com/1869202/174878341-0eca89cd-c4f4-4585-8b72-86a00d006c33.png">

**Comment by neilsf** [comment]
Hmmm... if you try `PRINT s, a` you'll get a big mess which means there's a problem with reserving strings on the stack frame. I'll fix that.

**Comment by neilsf** [comment]
Fixed in v3.1 alpha.
